docs/python/tutorials/workflow-tutorial.md

```diff
@@ -21,24 +21,30 @@ def reminder_workflow(email: str, time_to_sleep: int):
 
 Here are some example apps demonstrating what workflows can do:
 
-- [**Widget Store**](../examples/widget-store.md): No matter how many times you crash this online storefront, it always correctly processes your orders.
+- [**Fault-Tolerant Checkout**](../examples/widget-store.md): No matter how many times you crash this online storefront, it always correctly processes your orders.
 - [**Scheduled Reminders**](../examples/scheduled-reminders.md): Send a reminder email to yourself on any day in the future&mdash;even if it's months away.
+- [**Document Ingestion Pipeline**](../examples/document-detective.md): Use workflows and [queues](./queue-tutorial.md) to reliably process thousands of documents concurrently.
 
 
 ## Reliability Guarantees
 
 Workflows provide the following reliability guarantees.
 These guarantees assume that the application and database may crash and go offline at any point in time, but are always restarted and return online.
 
-1.  Workflows always run to completion.  If a DBOS process crashes while executing a workflow and is restarted, it resumes the workflow from the last completed step.
+1.  Workflows always run to completion.  If a DBOS process is interrupted while executing a workflow and restarts, it resumes the workflow from the last completed step.
 2.  [Steps](./step-tutorial.md) are tried _at least once_ but are never re-executed after they complete.  If a failure occurs inside a step, the step may be retried, but once a step has completed, it will never be re-executed.
 3.  [Transactions](./transaction-tutorial.md) commit _exactly once_.  Once a workflow commits a transaction, it will never retry that transaction.
 
+If an exception is thrown from a workflow, the workflow is **considered complete**&mdash;DBOS records the exception, sets the workflow status to `ERROR`, and **does not recover the workflow**.
```

Comment:
I think "complete" is unclear. Complete sounds like it ran through all the steps. It is errored and stopped in the middle.